### PR TITLE
[NV-1812] Email Webhook Provider

### LINKED ISSUE #2958: [NV-1812] Webhook Provider
### Why? (Context)

For companies that have legacy delivery systems, novu can trigger an API request with all the information related to the trigger to a webhook URL, the novu customer will be responsible for the actual delivery of the message.

### What?

Implementing a new Webhook provider for each channel type novu has so that the content will still be managed on Novu's side, and instead of calling a provider such as SendGrid, we will perform an API call to the webhook.

### Definition of Done

* Webhook provider for each channel type we have
* Users can specify the webhook URL when sending a message
* Novu will attach an HMAC key with the payload so the user can verify that novu is the initiator of the call
* Novu will retry to call the webhook in case it's down
* A payload will be sent to the webhook to include the information about: notification, subscriber, and trigger payload.

<sub>From [SyncLinear.com](https://synclinear.com) | [NV-1812](https://linear.app/novu/issue/NV-1812/webhook-provider)</sub>

### PR DESCRIPTION
**Pull request by jlucidar**
### What change does this PR introduce?

Adds a webhook provider for the following channels :
- email

### Why was this change needed?

Closes #2958

### Other information (Screenshots)
image

image

### CODE REVIEW

**Comment by scopsy** [comment]
@jlucidar incredible work! Will be reviewing this with the team soon!

**Comment by jlucidar** [comment]
As I stated in the linked Issue, this is still a WIP as it only concerns the email channel ( but it would already fit my company needs 😄 )

**Comment by jlucidar** [comment]
Any news on this ?
As this PR is similar to https://github.com/novuhq/novu/pull/3266 , it should be straightforward to review :)
Let me know if I can be of any help !

File: apps/worker/src/app/workflow/usecases/send-message/send-message-email.usecase.ts [hunk 1]
```diff
@@ -244,6 +244,11 @@ export class SendMessageEmail extends SendMessageBase {
       mailData.replyTo = command.overrides?.email?.replyTo as string;
     }
 
+    if (integration.providerId === 'email-webhook') {
```
Comment (LetItRock) [inline apps/worker/src/app/workflow/usecases/send-message/send-message-email.usecase.ts:247]: please use the `EmailProviderIdEnum.EmailWebhook`

**Review by LetItRock — APPROVED** [review]
looks good to me! 🚀 thanks for your contribution 🙌

**Comment by jlucidar** [comment]
Hi, any chance this get merged in the upcoming days ? Tell me if I can do something to get this forward.

**Comment by scopsy** [comment]
Surething @jlucidar would just need the testing pipeline to pass and we can merge, trying right now updating the lock file. I think it is related to it 🙏

**Comment by jlucidar** [comment]
> Surething @jlucidar would just need the testing pipeline to pass and we can merge, trying right now updating the lock file. I think it is related to it 🙏

Yup, could be related to this, I had several merge conflicts with this file.

**Comment by jlucidar** [comment]
It seems to be related to the newrelic package that fails to build : https://github.com/novuhq/novu/actions/runs/5221485760/jobs/9425884315?pr=3244#step:5:2073
Tried a few thing to fix the api docker build locally and using a base image with `make` command helped (using the same image you used but based on ubuntu instead of alpine)

**Comment by scopsy** [comment]
Alright it now okey, the docker pipeline problem is not related to your changes but only happens on Fork PR's for some reason :|

merging this, thank you so much for the contribution!!! <3

**Comment by jainpawan21** [comment]
@jlucidar This change will be included in 0.17.0 release.
We deployed 0.16.0 version before this PR merge.
Unfortunately, We were late in release notes

**Comment by jlucidar** [comment]
> @jlucidar This change will be included in 0.17.0 release.
We deployed 0.16.0 version before this PR merge.
Unfortunately, We were late in release notes

Great ! (just saw it was actually merged in 0.16.1)

**Pull request merged**